lib/controllers/situations.js

```diff
@@ -49,93 +49,76 @@ exports.create = function(req, res, next) {
+        foyers_fiscaux: testCase.foyers_fiscaux,
         individus: testCase.individus,
-        familles: testCase.familles[0],
-        foyers_fiscaux: testCase.foyers_fiscaux[0],
-        menages: testCase.menages[0]
+        menages: testCase.menages,
+        period: openfiscatest.scenarios[0].period,
     });
+    return test;
 }
 
-function myToString(obj) {
+function toStringOf(obj) {
     return obj.toString();
 }
 
-function normalizeTestContent(testFileContent) {
-    testFileContent.individus = testFileContent.individus.map(function(individu) {
-        individu.id = individu.id.toString();
-        return individu;
-    });
-
-    function replaceElementsWithStringRepresentation(obj) {
-        function convert(leafName) {
-            obj[leafName] = obj[leafName].map(myToString);
-        }
-        return convert;
-    }
-
-    var sources = {
-        familles: ['parents', 'enfants'],
+function normalizeIDs(test) {
+    var ID_PROPERTIES = {
+        individus: ['id'],
+        familles: ['enfants', 'parents'],
         foyers_fiscaux: ['declarants', 'personnes_a_charge'],
-        menages: ['enfants'],
+        menages: ['conjoint', 'enfants', 'personne_de_reference'],
     };
 
-    for (var parent in sources) {
-        sources[parent].forEach(replaceElementsWithStringRepresentation(testFileContent[parent]));
-    }
-
-    testFileContent.menages.personne_de_reference = testFileContent.menages.personne_de_reference.toString();
-    if (testFileContent.menages.conjoint) {
-        testFileContent.menages.conjoint = testFileContent.menages.conjoint.toString();
-    }
+    Object.keys(ID_PROPERTIES).forEach(function(entity) {
+        test[entity].forEach(function(value, index) {
+            ID_PROPERTIES[entity].forEach(function(property) {
+                if (test[entity][index][property] instanceof Array)
+                    test[entity][index][property] = test[entity][index][property].map(toStringOf);
+                else if (test[entity][index][property])
+                    test[entity][index][property] = test[entity][index][property].toString();
+            });
+        });
+    });
 }
 
-function createTestInYAML(situation, test) {
-    var testFileContent = createTestContent(situation, test);
-    normalizeTestContent(testFileContent);
-
-    return require('json2yaml').stringify(testFileContent);
+function toYAML(test) {
+    normalizeIDs(test);
+    return require('js-yaml').safeDump(test);
 }
 
-exports.openfiscaTest = function(req, res, next) {
-    var test = {
-        name: req.query.name || req.body.name || 'Missing name',
-        description: req.query.description || req.body.description || 'Missing description',
-        output_variables: req.query.expectedResults || req.body.expectedResults,
-    };
-    if (test.output_variables)
-        return res.send(createTestInYAML(req.situation, test));
+exports.openfiscaTest = function openfiscaTest(req, res) {
+    var DECORATION_ATTRIBUTES = [
```

Comment:
If this is a constant, it would be better to declare it outside from the scope of this function, as you are re-instantiating a new Array on each call.